pkg/pillar/cmd/nodeagent/nodeagent.go

```diff
@@ -752,6 +774,44 @@ func handleVaultStatusImpl(ctxArg interface{}, key string,
 	}
 }
 
+func handleVolumeMgrStatusCreate(ctxArg interface{}, key string,
+	statusArg interface{}) {
+	handleVolumeMgrStatusImpl(ctxArg, key, statusArg)
+}
+
+func handleVolumeMgrStatusModify(ctxArg interface{}, key string,
+	statusArg interface{}, oldStatusArg interface{}) {
+	handleVolumeMgrStatusImpl(ctxArg, key, statusArg)
+}
+
+func handleVolumeMgrStatusImpl(ctxArg interface{}, key string,
+	statusArg interface{}) {
+
+	ctx := ctxArg.(*nodeagentContext)
+	vms := statusArg.(types.VolumeMgrStatus)
+
+	if vms.RemainingSpace == 0 {
+		log.Warnf("MaintenanceMode due to no remaining diskspace")
+		// No not overwrite a vault maintenance mode
+		if !ctx.maintMode {
+			log.Noticef("Setting %s",
+				types.MaintenanceModeReasonNoDiskSpace)
+			ctx.maintModeReason = types.MaintenanceModeReasonNoDiskSpace
+			ctx.maintMode = true
+		}
+	} else {
+		// Do we need to clear maintenance?
+		if ctx.maintMode &&
+			ctx.maintModeReason == types.MaintenanceModeReasonNoDiskSpace {
+			log.Noticef("Clearing %s",
+				types.MaintenanceModeReasonNoDiskSpace)
+			ctx.maintMode = false
+			ctx.maintModeReason = types.MaintenanceModeReasonNone
+		}
+	}
+	publishNodeAgentStatus(ctx)
```

Comment:
As I understand it's also possible here that nothing is changed from the status perspective. Why publish the status then?

Reply:
Yeah - being sloppy (pubsub actually does a comparison and doesn't send to subscribers if there is no change, but the pattern we use is to have a changed boolean). Will fix.